### ISSUE TITLE
Extension crashes visual studio on "Save All"

### ISSUE THREAD

**Issue by bigworld12** [Xavalon/XamlStyler#216]
Having this extension installed in vs 2019, causes [this](https://developercommunity.visualstudio.com/content/problem/794811/visual-studio-crashes-when-pressing-save-all-while.html) issue.

> 1) Create a brand new wpf .net core project
> 2) open MainWindow.xaml
> 3) open MainWindow.xaml.cs
> 4) make any edit in MainWindow.xaml (e.g. adding a space character somewhere)
> 5) press “Save All”
> 6) VS crashes and can only be closed via task manager
>
> I suspect this to be some deadlock issue that happens when having the code-behind and the xaml of a user control open at the same time

**Comment by bigworld12** [comment]
it appears that the problem is probably originating from somewhere here 
https://github.com/Xavalon/XamlStyler/blob/master/XamlStyler.Package/StylerPackage.cs#L119

**Comment by grochocki** [comment]
I am unable to repro this issue with these steps on VS2019 16.2.5 with the latest XAML Styler release (3.2.5) or the source built from master. What versions are you on?

**Comment by tmatz** [comment]
Following exception can be observed by debugging XamlStyler.

XamlStyler uses parallel execution for 'Save All', but some method call is not allowed in this situation.

Maybe this problem is related.

```
System.Reflection.TargetInvocationException
  HResult=0x80131604
  Message=Exception has been thrown by the target of an invocation.
  Source=mscorlib
  StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments) in f:\dd\ndp\clr\src\BCL\system\reflection\methodinfo.cs:line 761

Inner Exception 1:
AggregateException: One or more errors occurred.

Inner Exception 2:
InvalidOperationException: Due to high risk of deadlock you cannot call GetService from a background thread in an AsyncPackage derived class. You should instead call GetServiceAsync (without calling Result or Wait on the resultant Task object) or switch to the UI thread with the JoinableTaskFactory.SwitchToMainThreadAsync method before calling GetService.
```

**Comment by grochocki** [comment]
Hmm, we _are_ doing what is suggested in the exception details with GetServiceAsync + JoinableTaskFactory.SwitchToMainThreadAsync (e.g., in[InitializeAsync](https://github.com/Xavalon/XamlStyler/blob/949f32b7cf0e3e5d4f4f802a1085c084477cef54/XamlStyler.Package/StylerPackage.cs#L47)), or are you suggesting it is something similar to that? Either way, it is hard for me to debug if I cannot observe the crash.

**Comment by tmatz** [comment]
I've fixed this problem in my repository. Please check this
https://github.com/tmatz/XamlStyler/commit/fdef3d6fd835b45cbe9b2d738b50bc4593f2147a

**Comment by grochocki** [comment]
Should be fixed via #220 now.

**Comment by bigworld12** [comment]
I can still reproduce this on v3.1912.9
